tests/components/sensor/test_template.py

```diff
@@ -0,0 +1,44 @@
+"""
+tests.components.sensor.template
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Tests template sensor.
+"""
+from unittest.mock import patch
+
+import pytest
+
+import homeassistant.core as ha
+import homeassistant.components.sensor as sensor
+
+
+class TestSensorYr:
+    """ Test the Yr sensor. """
+
+    def setup_method(self, method):
+        self.hass = ha.HomeAssistant()
+
+    def teardown_method(self, method):
+        """ Stop down stuff we started. """
+        self.hass.stop()
+
+    def test_template(self, betamax_session):
+        assert sensor.setup(self.hass, {
+            'sensor': {
+                'platform': 'template',
+                'sensors': {
+                    'test_template_sensor': {
+                        'value_template':
+                            '{{ states.sensor.test_state.state }}'
+                    }
+                }
+            }
+        })
+
+        state = self.hass.states.get('sensor.test_template_sensor')
+        assert state.state == ''
```

Comment:
I don't want to be more strict, that will only cause annoyance.

Instead of setting it to unknown , let's set it to `error` when an exception occurs and let's also log the text (not the whole exception)